### PR TITLE
Update package

### PR DESCRIPTION
**Pull request by naacbin**
Update/fix various packages.

I have run all packages except IDA plugins with the new version of IDA.

### CODE REVIEW

**Comment by naacbin** [comment]
The CLI fail because `memprocfs.vm` is installed after `arsenalimagemounter.vm` which already install dokan driver within the ZIP.

**Review by Ana06 — CHANGES_REQUESTED** [review]
Thanks a lot for fixing these packages @naacbin!

> The CLI fail because memprocfs.vm is installed after arsenalimagemounter.vm which already install dokan driver within the ZIP.

We need to ensure that all packages install correctly independently on the order and on which other packages are installed. Even if the error was not introduced in this PR, we need to fix it in this PR or prior to merge this PR because if the CI fails, we can't merge the PR and the packages won't be pushed to MyGet by our CI.

Two ideas to fix the issue:
- If `dokan.vm` provides the driver we want to install in `arsenalimagemounter.vm`, remove the ZIP instalation and require `dokan.vm` as dependency in  `arsenalimagemounter.vm`
- Install the ZIP with `-ErrorAction SilentlyContinue` so that it doesn't fail if it is already installed and check that it there afterwards with `VM-Assert-Path`

File: packages/hayabusa.vm/tools/chocolateyinstall.ps1
Comment (naacbin) [reply packages/hayabusa.vm/tools/chocolateyinstall.ps1:7]: I did it this way so that it could be updated automatically by the pipeline (but I can be wrong)

File: packages/x64dbg.vm/tools/chocolateyinstall.ps1
Comment (naacbin) [reply packages/x64dbg.vm/tools/chocolateyinstall.ps1:11]: I think it's better to use a full URL as there is only one place where the variable is employed and that's the URL. Moreover, this is the only package that does this (along with `hayabusa.vm` but it's different), so I think we should remove `$snapshotDate`.

File: packages/hayabusa.vm/tools/chocolateyinstall.ps1
Comment (Ana06) [reply packages/hayabusa.vm/tools/chocolateyinstall.ps1:7]: It won't because the executable name also includes the version. But we could modify the `update_package.py script to replace the version in the whole `chocolateyinstall.ps1` text instead of only in the `url` and this installer to use the whole executable name:
`$executableName = `hayabusa-2.15.0-win-x64.exe"`

Then the CI should update the package.

I am ok with this change, even if we do not implement this change in this PR.

**Review by Ana06 — CHANGES_REQUESTED** [review]
@naacbin 

> I have added to arsenalimagemounter.vm, dokan.vm dependency as the ZIP is required for other drivers.

I do not understand this fix. If we need to install the ZIP, why do we need dokan as dependency and how does this fix the failure?

I think the file `windbg.msixbundle` added in this PR was added by accident (likely when testing https://github.com/mandiant/VM-Packages/issues/1040). Can you please remove it @naacbin?

**Comment by naacbin** [comment]
> @naacbin thanks for the changes and explanations!
> 
> > I have added to arsenalimagemounter.vm, dokan.vm dependency as the ZIP is required for other drivers.
> 
> I do not understand this fix. If we need to install the ZIP, why do we need dokan as dependency and how does this fix the failure?
> 
> I think the file `windbg.msixbundle` added in this PR was added by accident (likely when testing #1040). Can you please remove it @naacbin?

The failure come from the fact that the install of dokan is done via MSI package. When an MSI package detect an already installed program, he won't try to "reinstall" it (you can see that with the GUI). The ZIP contains the dokan driver (as well as other) but does not install it via MSI meaning it won't fail. So by adding the dokan dependency to `arsenalimagemounter.vm` it will first install dokan via MSI and then the ZIP which doesn't fail. I hope it's clear.

Thanks, I push the file without paying attention.

File: packages/idafree.vm/tools/chocolateyuninstall.ps1
Comment (naacbin) [reply packages/idafree.vm/tools/chocolateyuninstall.ps1:13]: Should I remove this from the commit to push in another PR ?

File: packages/hayabusa.vm/tools/chocolateyinstall.ps1
Comment (naacbin) [reply packages/hayabusa.vm/tools/chocolateyinstall.ps1:7]: It's the same as leaving as modified and editing  `update_package.py` to replace all occurrence of the version ? I can do it in this PR :)

File: packages/idafree.vm/idafree.vm.nuspec
Comment (naacbin) [reply packages/idafree.vm/idafree.vm.nuspec:5]: If needed, I can remove this from this PR and do another (or an issue) separately to potentially update to the latest version.

File: packages/hayabusa.vm/tools/chocolateyinstall.ps1
Comment (Ana06) [reply packages/hayabusa.vm/tools/chocolateyinstall.ps1:7]: I would suggest leaving it as it is in this PR and create a new PR to modify `update_package.py` to replace all occurrence of the version so that we can review the two changes independently. :wink:

File: packages/idafree.vm/idafree.vm.nuspec
Comment (Ana06) [reply packages/idafree.vm/idafree.vm.nuspec:5]: after you took the effort to update it, I think we can keep the change.

File: packages/idafree.vm/tools/chocolateyuninstall.ps1
Comment (Ana06) [reply packages/idafree.vm/tools/chocolateyuninstall.ps1:13]: It is fine here! I just realised that we need to fix it in other places when reviewing it :wink:

**Comment by Ana06** [comment]
@naacbin thanks for the detailed explanation. Now I understand it.

> So by adding the dokan dependency to arsenalimagemounter.vm it will first install dokan via MSI and then the ZIP which doesn't fail.

I am ok with this solution! We may want to research if we can improve it, but everything works with this setup.

**Review by Ana06 — APPROVED** [review]

**Comment by Ana06** [comment]
Thanks for all the work @naacbin!

**Comment by naacbin** [comment]
@Ana06, I have check `update_package.py` it already seems to replace all version strings in a file thankfully to [this line](158b404fa5fd6937a1331ed1acde262998e6e1586a8604346956d4fc6a14b5d6) (present in both `update_version_url` and `update_github_url` function).
However, `hayabusa` as well as other packages such as `upx` have a problem when the version for example `v2.15.0` is different that the version in the filename `hayabusa-2.15.0-win-x64.zip` because of the `v`. The update works only because it will fallback to `update_github_url` and the updated version is in one of the case of `get_increased_version`.

From my testing adding `v?` in the url of `update_github_url` will allow the update via github without breaking any other package auto updates (but I think a deeper review is required)  :
 - `download/v?(?P<version>[^/]+)/`
 - `tags/v?(?P<version>[^/]+).zip`
 
 I can make a PR for this change.

**Pull request merged**
